modules/core/src/getDeviceDisclosures.ts

```diff
@@ -0,0 +1,114 @@
+import {DeviceDisclosure} from './model';
+import {Json} from './Json';
+import {DeviceDisclosureError} from './errors';
+
+/**
+ * Processeses a device disclosure object according to IAB specifications.
+ * @param {unknown} objects - Raw object that has device disclosure data.
+ * @return {DeviceDisclosure} - Returns an array of processed DeviceDisclosure objects.
+ */
+const processDeviceDisclosure = (objects: unknown): DeviceDisclosure[] => {
+
+  const dirtyDeviceDisclosures = Object.keys(objects).map((key) => objects[key]);
+  const validKeys: string[] = [
+    'domain',
+    'cookieRefresh',
+    'identifier',
+    'maxAgeSeconds',
+    'purposes',
+    'type',
+  ];
+
+  return dirtyDeviceDisclosures
+    .map((dirtyDeviceDisclosure): DeviceDisclosure => {
+
+      /*
+       * Create a clean DeviceDisclosure object with default values. */
+      const cleanedDeviceDisclosure: DeviceDisclosure = {
+        identifier: '',
+        maxAgeSeconds: null,
+        purposes: [],
+        type: '',
+      };
+
+      /*
+       * We process the object by only taking the values with a valid key and assigning it to
+       * our clean DeviceDisclosure object. */
+      validKeys.forEach((validKey: string): void => {
+
+        if (dirtyDeviceDisclosure[validKey]) {
+
+          cleanedDeviceDisclosure[validKey] = dirtyDeviceDisclosure[validKey];
+
+        }
+
+      });
+
+      /*
+      * Only display the domain if disclosure is of type 'web' or 'cookie'. */
+      if (cleanedDeviceDisclosure.type == 'app' && cleanedDeviceDisclosure.domain) {
+
+        cleanedDeviceDisclosure.domain = undefined;
+
+      }
+
+      /*
+       * Only display maxAgeSeconds and cookieRefresh if disclosure is of type 'cookie'. */
+      if (cleanedDeviceDisclosure.type == 'app' || cleanedDeviceDisclosure.type == 'web') {
+
+        if (cleanedDeviceDisclosure.maxAgeSeconds !== null) {
+
+          cleanedDeviceDisclosure.maxAgeSeconds = null;
+
+        }
+
+        if (cleanedDeviceDisclosure.cookieRefresh !== undefined) {
+
+          cleanedDeviceDisclosure.cookieRefresh = undefined;
+
+        }
+
+      }
+
+      return cleanedDeviceDisclosure;
+
+    })
+    /*
+     * Any objects that have default values should be filtered out of the array.  The case being
```

Comment:
What are we gaining by cleaning unknown keys and removing default values? I'm wondering why we don't just return the disclosure object as-fetched (as that will be automatically extensible instead of requiring code changes).

Reply:
Hello, so the purpose of cleaning the object was to make sure that this function that the library could provide, returns an object that adhered to the specifications linked at the PR description.

And this is entirely optional to use, so if any users wanted the as-is fetched object then they can perform their own fetching and processing.  We basically just thought that it would be nice to share the code we are using to fetch and processes, since other people could be using the same process.

Reply:
If you guys think that this is unnecessary, totally cool.  Just lemme know so I can just edit the PR for the DeviceDisclosure interface and Vendor interface changes.

Reply:
I agree with @a2intl from my experience maintaining this library, you want as little code out there as possible to accomplish the goal, because everything that is written will need to be maintained + unit tested + documented + supported in GitHub issues.  If a user can just use the `Json.fetch` function then that is probably good enough.

Reply:
alright sounds good, I'll update the code.